src/app/digital/actions/FrequencyChangeAction.ts

```diff
@@ -0,0 +1,32 @@
+import {Action} from "core/actions/Action";
+
+import {Clock} from "digital/models/ioobjects/inputs/Clock";
+import {Oscilloscope} from "digital/models/ioobjects/outputs/Oscilloscope";
+
+
+export class FrequencyChangeAction implements Action {
+    private component: Clock | Oscilloscope;
+
+    private initialFreq: number;
+    private targetFreq: number;
+
+    public constructor(component: Clock | Oscilloscope, targetFreq: number) {
```

Comment:
Should there be a `type FrequencyComponent = Clock | Oscilloscope` or are the amount of uses for that too few?

Reply:
I'm probably gonna make a `TimedComponent` class that Clock and Oscilloscope will inherit from since they have quite the few similarities